src/webgpu/api/validation/createTexture.spec.ts

```diff
@@ -216,6 +216,86 @@ g.test('mipLevelCount,bound_check,bigger_than_integer_bit_width')
     });
   });
 
+g.test('sampleCount,various_sampleCount_with_all_formats')
+  .desc(`Test texture creation with various (valid or invalid) sample count and all formats`)
+  .subcases(() =>
+    params()
+      .combine(poptions('sampleCount', [1, 2, 4, 8, 16, 32, 256]))
+      .combine(poptions('format', kAllTextureFormats))
+  )
+  .fn(async t => {
+    const { sampleCount, format } = t.params;
+    const descriptor = {
+      size: [32, 32, 1],
+      sampleCount,
+      format,
+      usage: GPUTextureUsage.SAMPLED,
+    };
+
+    await t.selectDeviceOrSkipTestCase(kAllTextureFormatInfo[format].extension);
+
+    const success =
+      sampleCount === 1 || (sampleCount === 4 && format in kUncompressedTextureFormats);
```

Comment:
I intended to use the code snippet to say that it will success if: 
1) sample count is 1, or
2) sample count is 4 and the format is not compressed texture formats. 

But looks like it is not consistent with my intention. 

Is there a way to debug the script code?

Reply:
You can use the standard JS debugger in Chrome dev tools. If you are using `npm run dev` then it will source-map the generated JS back to TypeScript.

You can use the little "log stack to console" button ![image](https://user-images.githubusercontent.com/606355/109887669-fecf2a80-7c36-11eb-80ee-377b80051e26.png) on a test or log line to print a stack you can use to jump to the test code.

The issue here is that `in` checks on object properties, not array elements; you have to use `Array.includes`:
```suggestion
      sampleCount === 1 || (sampleCount === 4 && kUncompressedTextureFormats.includes(format));
```

Reply:
It leads to type mismatch between kUncompressedTextureFormats and format. And I can't use `as` here. However, if I use a user-defined constant exactly the same with kUncompressedTextureFormats, it will ignore the type check. It is like a workaround. Maybe there are better solutions.

I really think the type check between kUncompressedTextureFormats and format is not necessary. Otherwise, `includes` is not effective as it supposed to be (if they are exactly the same, why we name it `includes`. `include` means a set and a subset).

Reply:
> You can use the standard JS debugger in Chrome dev tools. If you are using `npm run dev` then it will source-map the generated JS back to TypeScript.
> 
> You can use the little "log stack to console" button ![image](https://user-images.githubusercontent.com/606355/109887669-fecf2a80-7c36-11eb-80ee-377b80051e26.png) on a test or log line to print a stack you can use to jump to the test code.

Thank you, Kai.